federation/src/server/plugins.ts

```diff
@@ -23,8 +23,8 @@ const setHeadersPlugin = {
 
 const filterVariables = (variables: any) => {
   const vars = clonedeep(variables)
-  if (vars.password) vars.password = '***'
-  if (vars.passwordNew) vars.passwordNew = '***'
+  if (vars && vars.password) vars.password = '***'
+  if (vars && vars.passwordNew) vars.passwordNew = '***'
```

Comment:
Could be replaced by this or not?
```suggestion
  if (vars?.password) vars.password = '***'
  if (vars?.passwordNew) vars.passwordNew = '***'
```

Reply:
If its ok I'd like to postpone this change till we have the typescript strict rule enforcing it:
- https://typescript-eslint.io/rules/prefer-optional-chain

Till we have this I'd like to keep backend and federation code equal (those respective lines are identical as it currently is)